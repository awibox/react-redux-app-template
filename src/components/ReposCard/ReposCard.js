import React from 'react';
import PropTypes from 'prop-types';
// Components
import { Card } from 'tigerspack';
// Styles
import styles from './ReposCard.scss';

// Using "Stateless Functional Components"
const ReposCard = (props) => {
  const { repo, style } = props;
  return (
    <div className={styles.repo}>
      {repo.id > 0
        && <Card theme={'light'}>
          <a href={repo.html_url} rel="noopener noreferrer" target="_blank" className={styles.name}>{repo.name}</a>
          <div className={styles.githubIcon}><i className="fa fa-github" aria-hidden="true"></i></div>
          <div className={styles.desc}>{repo.description}</div>
          <div className={styles.panel}>
            {repo.language
            && <div className={styles.property}><i className="fa fa-circle" style={style}></i> {repo.language}</div>}
            <div className={styles.size}><i className="fa fa-file-archive-o"></i> {(repo.size / 1024).toFixed(2)} mb
            </div>
            <div className={styles.stars}><i className="fa fa-star" aria-hidden="true"></i> {repo.stargazers_count}
            </div>
          </div>
        </Card>
      }
    </div>
  );
};

ReposCard.propTypes = {
  repo: PropTypes.shape({
    html_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    language: PropTypes.string,
    id: PropTypes.number.isRequired,
    size: PropTypes.number,
    stargazers_count: PropTypes.number,
  }),
  style: PropTypes.object,
};

export default ReposCard;
