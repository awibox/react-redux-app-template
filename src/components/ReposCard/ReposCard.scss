@import "../../styles/variables";

.repo {
  padding-left: $defaultIndent;
  position: relative;
  width: 20%;
  box-sizing: border-box;
}

.name {
  color: #4078c0;
  font-size: 16px;
}

.desc {
  font-size: 14px;
  margin-bottom: 16px;
  margin-top: 8px;
  color: #767676;
  font-weight: 100;
  min-height: 18px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.panel {
  display: flex;
  flex-flow: row;
}

.property {
  color: #767676;
  font-weight: 100;
  margin-right: 50px;
}

.fa {
  margin-right: 5px;
}

.stars {
  @extend .property;
  flex: 1;
  text-align: right;
  margin-right: 0;
  .fa {
    margin-right: 0;
  }
}

.size {
  @extend .property;
}

.githubIcon {
  font-size: 30px;
  position: absolute;
  top: 10px;
  right: 10px;
}
@media(max-width: $widthScreenDisplayLarge) {
  .repo {
    width: 25%;
  }
}
@media(max-width: $widthScreenDisplayDefault) {
  .repo {
    width: 33.33%;
  }
}
@media(max-width: $widthScreenLaptopSmall) {
  .repo {
    width: 50%;
  }
}
@media(max-width: $widthScreenTabletPortrait) {
  .repo {
    width: 100%;
  }
}

