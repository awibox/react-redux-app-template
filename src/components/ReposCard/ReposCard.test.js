import React from 'react';
import { shallow } from 'enzyme';
import ReposCard from './ReposCard';

const mockData = {
  html_url: 'http://localhost:8888/item',
  name: 'First Repo',
  description: 'Repository',
  language: 'javascript',
  id: 1,
  size: 10,
  stargazers_count: 10,
};
const mockStyle = {
  background: 'red',
};

describe('ReposCard', () => {
  it('should render correctly', () => {
    const component = shallow(<ReposCard repo={mockData}/>);
    expect(component).toMatchSnapshot();
  });
  it('should have style', () => {
    const component = shallow(<ReposCard repo={mockData} style={mockStyle}/>);
    expect(component.find('.fa-circle')).toHaveStyle(mockStyle);
  });
});
