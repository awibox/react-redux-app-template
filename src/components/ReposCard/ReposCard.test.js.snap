// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposCard should render correctly 1`] = `
<div
  className="repo"
>
  <Card
    indent={15}
    padding={15}
    theme="light"
  >
    <a
      className="name"
      href="http://localhost:8888/item"
      rel="noopener noreferrer"
      target="_blank"
    >
      First Repo
    </a>
    <div
      className="githubIcon"
    >
      <i
        aria-hidden="true"
        className="fa fa-github"
      />
    </div>
    <div
      className="desc"
    >
      Repository
    </div>
    <div
      className="panel"
    >
      <div
        className="property"
      >
        <i
          className="fa fa-circle"
        />
         
        javascript
      </div>
      <div
        className="size"
      >
        <i
          className="fa fa-file-archive-o"
        />
         
        0.01
         mb
      </div>
      <div
        className="stars"
      >
        <i
          aria-hidden="true"
          className="fa fa-star"
        />
         
        10
      </div>
    </div>
  </Card>
</div>
`;
